charts/minecraft/templates/extraports-ing.yaml

```diff
@@ -1,6 +1,6 @@
 {{- $minecraftFullname := include "minecraft.fullname" . }}
 {{- range .Values.minecraftServer.extraPorts }}
-{{- if default "" .ingress.enabled }}
+{{- if default "" (.ingress).enabled }}
```

Comment:
Very interesting. So this allows for nil-safe access to objects? So you happen to know where in the helm docs they describe that?

Reply:
I actually found this while poking through Google, with [this StackOverflow post](https://stackoverflow.com/a/77633748/19815719) leading me to the right answer. 

>In Helm we can wrap a value that we might consider optional in brackets.
>
> `type: {{ (.Values.fpm.service).type | default "NodePort" }}`
>
>Now when we try to run a helm template / install, it will consider .Values.fpm.service to be optional. We would still run into a Nil pointer if .Values.fpm is not set. So you must ensure to encapsulate up until where should be considered optional, as you can see in the example I provided, I did not place brackets around .type otherwise, it would try to find service which may not exist.

Digging into this further, [another StackOverflow post](https://stackoverflow.com/a/68807258/19815719) notes that this behavior comes from Go's `text/template` library. It seems that something like `{{ if default "" .ingress.enabled }}` is called a "pipeline" and adding the parenthesis makes a pipeline within the pipeline, like nested operators `if default "" (.ingress).enabled`. The nested pipeline returns a nil value under a suite of negative circumstances, which is then used instead of a nil pointer when accessing `.enabled` ([docs](https://pkg.go.dev/text/template#hdr-Actions)).

> If the value of the pipeline is empty, no output is generated
> ...
> The empty values are false, 0, any nil pointer or interface value, and any array, slice, map, or string of length zero.

Very interesting!